src/fparser/tests/fparser2/test_Fortran2003.py

```diff
@@ -3215,53 +3340,105 @@ def test_Module_Subprogram_Part(): # R1107
 
-    a = cls('use :: a, operator(.hey.)=>operator(.hoo.)')
-    assert isinstance(a, cls),repr(a)
-    assert_equal(str(a),'USE :: a, OPERATOR(.HEY.) => OPERATOR(.HOO.)')
+    inst = ucls('use :: a, only: b')
+    assert isinstance(inst, ucls), repr(inst)
+    assert str(inst) == 'USE :: a, ONLY: b'
+    assert repr(inst) == (
+        "Use_Stmt(None, '::', Name('a'), ', ONLY:', Name('b'))")
 
-    a = cls('use, intrinsic :: a, operator(.hey.)=>operator(.hoo.), c=>g')
-    assert isinstance(a, cls),repr(a)
-    assert_equal(str(a),'USE, INTRINSIC :: a, OPERATOR(.HEY.) => OPERATOR(.HOO.), c => g')
+    inst = ucls('use a, ONLY : b')
+    assert isinstance(inst, ucls), repr(inst)
+    assert str(inst) == 'USE a, ONLY: b'
+    assert repr(inst) == (
+        "Use_Stmt(None, None, Name('a'), ', ONLY:', Name('b'))")
 
-def test_Module_Nature(): # R1110
+    inst = ucls('use, intrinsic :: a, ONLY: b')
+    assert isinstance(inst, ucls), repr(inst)
+    assert str(inst) == 'USE, INTRINSIC :: a, ONLY: b'
+    assert repr(inst) == (
+        "Use_Stmt(Module_Nature('INTRINSIC'), '::', Name('a'), "
+        "', ONLY:', Name('b'))")
+
+    inst = ucls('use, non_intrinsic :: a, ONLY: b, c, d')
+    assert isinstance(inst, ucls), repr(inst)
+    assert str(inst) == 'USE, NON_INTRINSIC :: a, ONLY: b, c, d'
+    assert repr(inst) == (
+        "Use_Stmt(Module_Nature('NON_INTRINSIC'), '::', Name('a'), "
+        "', ONLY:', Only_List(',', (Name('b'), Name('c'), Name('d'))))")
+
+    inst = ucls('use a, c=>d')
+    assert isinstance(inst, ucls), repr(inst)
+    assert str(inst) == 'USE a, c => d'
+    assert repr(inst) == (
+        "Use_Stmt(None, None, Name('a'), "
+        "',', Rename(None, Name('c'), Name('d')))")
+
+    inst = ucls('use :: a, operator(.hey.)=>operator(.hoo.)')
+    assert isinstance(inst, ucls), repr(inst)
+    assert str(inst) == 'USE :: a, OPERATOR(.HEY.) => OPERATOR(.HOO.)'
+    assert repr(inst) == (
+        "Use_Stmt(None, '::', Name('a'), ',', "
+        "Rename('OPERATOR', Defined_Op('.HEY.'), Defined_Op('.HOO.')))")
+
+    inst = ucls('use, intrinsic :: a, operator(.hey.)=>operator(.hoo.), c=>g')
+    assert isinstance(inst, ucls), repr(inst)
+    assert str(inst) == (
+        'USE, INTRINSIC :: a, OPERATOR(.HEY.) => OPERATOR(.HOO.), c => g')
+    assert repr(inst) == (
+        "Use_Stmt(Module_Nature('INTRINSIC'), '::', Name('a'), "
+        "',', Rename_List(',', ("
+        "Rename('OPERATOR', Defined_Op('.HEY.'), Defined_Op('.HOO.')), "
+        "Rename(None, Name('c'), Name('g')))))")
+
+    inst = ucls('use, non_intrinsic :: a, ONLY: b => c')
+    assert isinstance(inst, ucls), repr(inst)
+    assert str(inst) == 'USE, NON_INTRINSIC :: a, ONLY: b => c'
+    assert repr(inst) == (
+        "Use_Stmt(Module_Nature('NON_INTRINSIC'), '::', Name('a'), "
+        "', ONLY:', Rename(None, Name('b'), Name('c')))")
+
+
+def test_Module_Nature():  # pylint: disable=invalid-name
```

Comment:
Please add a check that we don't get a match if something other than "intrinsic" or "non_intrinsic" is specified.

Reply:
Good point! Added, and I also added such a check to the `test_Private_Components_Stmt():`